### PR TITLE
Fix console errors

### PR DESCRIPTION
**Pull request by rachultimate**
-fix: Create a wrapper for Dialog component to fix `disableBackDropClick` deprecation errror.

Pull-Request for `paretOS`

## Description
When the page loaded, the browser was showing an `disableBackDropClick` error.
That was happening because the prop `disableBackDropClick` is deprecated and has been made redundant. To make it work, it's necessary to pass the funcion "onClose" together with the props.
After some searching, the solution that I found was creating a wrapper for the Dialog.
I tested it and it worked completely fine, no errors in the console.

## Relates to
- #142 

## Reviewers
- @jayeclark 

## Screenshots
Screenshot_1

## Notes
I'm sorry if I did anything wrong, that's my second time ever contributing to open-source projects in GitHub. Please tell me if I did anything wrong.

### CODE REVIEW

**Comment by jayeclark** [comment]
Hey @rachultimate , congratulations on your second ever OSS Pull Request and thanks for the contribution! This is a great start - great work!

There are a few things I noticed.

1. Re: the Dialog component in App.js, the modal actually covers the entire screen. It's not possible for a user to click on the backdrop, because none of the backdrop is visible... so it doesn't seem like disabling backdrop click is a necessary setting on that instance of the component. (No need for a custom wrapper, just remove the prop.)

2. However, your research & code wasn't in vain! There are still two other instances of Dialog that use disableBackdropClick and will throw the same error ([ExperienceModule](https://github.com/mikhael28/paretOS/blob/main/src/learn/ExperienceModule.js) and [LearnDashboard](https://github.com/mikhael28/paretOS/blob/main/src/learn/LearnDashboard.js)). We should probably also fix it in those files, if you'd like to apply the solution you found to either or both of those! 

3. Something else worth considering --- I found the article explaining the wrapper solution, but that workaround seems like it might be intended for a repository where there were many, many Dialog boxes using the disableBackdropClick prop, instead of just two. Adding an onClose callback directly to the Dialog component is the recommended way to replace the property [according to Material UI](https://newreleases.io/project/yarn/@material-ui/core/release/5.0.0-alpha.17).

**Relevant section**
Screen Shot 2022-03-20 at 2 33 16 PM

(The code snippet above is a _little_ misleading because it assumes there is an onClose callback defined somewhere else in your code. In this project you would replace that `onClose(event, reason)` with custom code relevant to that component.)

TLDR: disableBackdropClick in App.js can be removed from the Dialog component since there is no way to click on the backdrop. For the two remaining Dialog component instances that use disableBackdropClick (ExperienceModule and LearnDashboard), it may be simpler to replace the disableBackdropClick property with an inline onClose callback.

**Comment by rachultimate** [comment]
Thank you so much for taking time to consider my PR! I tried my best to try to find a solution for the error that was showing up in the console, and it was kinda hard since I'm not used to work with React using classes, I always use React Hooks instead, because I've heard that they are currently being more used than classes, but I can still work with classes as well.
Indeed, the article that I found mentioned that the wrapper solution was better to repositories with many Dialog boxes, but I decided to try it anyways to check if it could work fine as well, but by your explanation, now I understood and know exactly what I should've done! Thank you, wish you luck in your project :)

**Comment by jayeclark** [comment]
@rachultimate No need to close the PR! You can always add additional commits to revise it if you like. A PR is almost never perfect on the first go. 🙂 I didn't even notice the thing about the modal on the main App.js file not having a backdrop visible until I went to check that clicking on the backdrop would actually be disabled.

I hear you about class component vs hooks! (The LearnDashboard component that I mentioned actually has been converted to hooks.)
